rpcserver.go

```diff
@@ -2697,6 +2698,29 @@ func (r *rpcServer) PendingChannels(ctx context.Context,
 			LocalBalance:  int64(pendingClose.SettledBalance),
 		}
 
+		// Lookup the channel in the historical channel bucket to obtain
+		// initiator information. If the historical channel bucket was
+		// not found, or the channel itself, this channel was closed
+		// in a version before we started persisting historical
+		// channels, so we silence the error.
+		historical, err := r.server.chanDB.FetchHistoricalChannel(
+			&pendingClose.ChanPoint,
+		)
+		switch err {
+		// If the channel was closed in a version that did not record
+		// historical channels, ignore the error.
+		case channeldb.ErrNoHistoricalBucket:
+		case channeldb.ErrChannelNotFound:
+
+		case nil:
+			channel.Initiator = historical.IsInitiator
```

Comment:
Shouldn't the field distinguish between non-initiator and a legacy channel for which no history is present?

Reply:
hm yes it should, at the time I thought we wouldn't have legacy channels in pending, but there is of course the old upgrade path option of a channel closing before upgrade, and list pending after. This should use the `Initiator` enum.